src/utils/config.ts

```diff
@@ -1,63 +1,44 @@
 import { Configuration, ConfigurationParameters } from '../api';
-import { Config, EthNetwork } from '../types';
+import { Config } from '../types';
 import { version } from '../../package.json';
 
-const headers = { 'x-sdk-version': `imx-core-sdk-ts-${version}` };
+const defaultHeaders = { 'x-sdk-version': `imx-core-sdk-ts-${version}` };
 
-const mergeApiProperties = (
-  basePath: string,
-  apiConfigOptions: ConfigurationParameters,
-) => ({
-  basePath,
-  ...apiConfigOptions,
-  baseOptions: {
-    ...(apiConfigOptions.baseOptions || {}),
-    headers: {
-      ...(apiConfigOptions.baseOptions?.headers || {}),
-      ...headers,
-    },
-  },
-});
+type RequiredProperties<T, P extends keyof T> = Omit<T, P> &
+  Required<Pick<T, P>>;
 
-export const getConfig = (
-  network: EthNetwork = 'ropsten',
-  apiConfigOptions: ConfigurationParameters = {},
-): Config => {
-  switch (network) {
-    case 'dev':
-      return {
-        api: new Configuration(
-          mergeApiProperties(
-            'https://api.dev.x.immutable.com',
-            apiConfigOptions,
-          ),
-        ),
-        starkContractAddress: '0xd05323731807A35599BF9798a1DE15e89d6D6eF1',
-        registrationContractAddress:
-          '0x7EB840223a3b1E0e8D54bF8A6cd83df5AFfC88B2',
-      };
+const appendDefaultHeaders = (
+  apiConfigOptions: ConfigurationParameters,
+): ConfigurationParameters => {
+  apiConfigOptions.baseOptions = apiConfigOptions.baseOptions || {};
+  apiConfigOptions.baseOptions.headers = {
+    ...(apiConfigOptions.baseOptions.headers || {}),
+    ...defaultHeaders,
+  };
+  return apiConfigOptions;
+};
 
-    case 'ropsten':
-      return {
-        api: new Configuration(
-          mergeApiProperties(
-            'https://api.ropsten.x.immutable.com',
-            apiConfigOptions,
-          ),
-        ),
-        starkContractAddress: '0x4527BE8f31E2ebFbEF4fCADDb5a17447B27d2aef',
-        registrationContractAddress:
-          '0x6C21EC8DE44AE44D0992ec3e2d9f1aBb6207D864',
-      };
+interface ConfigParams {
+  starkContractAddress: string;
+  registrationContractAddress: string;
+  chainID: number;
+  apiConfigOptions: RequiredProperties<ConfigurationParameters, 'basePath'>;
+}
 
-    case 'mainnet':
-      return {
-        api: new Configuration(
-          mergeApiProperties('https://api.x.immutable.com', apiConfigOptions),
-        ),
-        starkContractAddress: '0x5FDCCA53617f4d2b9134B29090C87D01058e27e9',
-        registrationContractAddress:
-          '0x72a06bf2a1CE5e39cBA06c0CAb824960B587d64c',
-      };
+export const getConfig = ({
```

Comment:
Why not expose a few pre-defined environments e.g. 

```
const Production: Environment = {

}
```

Reply:
So you're saying hardcoding these addresses was never the issue? It was just the switch statement that needed to go?

Reply:
@alex-connolly we can expose a pre-configured `mainnet` config but `dev` doesn't make sense outside of IMX and `ropsten` is about to change.

Reply:
@zatlite the code inside of `appendDefaultHeaders` still looks scary - let's kill that function and just initialise `ConfigurationParameters` here with the default headers - will reduce LOC and improve readability.

Reply:
Agree @Kastet with not exposing dev - we should definitely get the new Goerli details, though

Reply:
@Kastet refactored append default headers
 [3ef2ccc](https://github.com/immutable/imx-core-sdk/pull/121/commits/3ef2ccc301e9f99409eac74a710154e9cbc8d6ae)

Reply:
It looks much cleaner now! 👍

Reply:
@alex-connolly we'll add Goerli details as soon as it's ready, should be in a couple of weeks.